File: delta/tests2/tests2.op.go
package tests2

import "github.com/dave/protod/delta"

func Op() Op_root_type {
	return Op_root_type{}
}

type Op_root_type struct{}

func (Op_root_type) Shirt() Shirt_type {
	return Shirt_type{}
}

type Shirt_type struct {
	location []*delta.Locator
}

func (b Shirt_type) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_type(l []*delta.Locator) Shirt_type {
	return Shirt_type{location: l}
}
func (b Shirt_type) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_type) Set(value *Shirt) *delta.Op {
	return delta.Set(b.location, value)
}
func (b Shirt_type) Designer() delta.String_scalar {
	return delta.NewString_scalar(delta.CopyAndAppendField(b.location, "designer", 1))
}
func (b Shirt_type) Size() delta.Uint32_scalar {
	return delta.NewUint32_scalar(delta.CopyAndAppendField(b.location, "size", 2))
}

type Shirt_list struct {
	location []*delta.Locator
}

func (b Shirt_list) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_list(l []*delta.Locator) Shirt_list {
	return Shirt_list{location: l}
}
func (b Shirt_list) Index(i int) Shirt_type {
	return NewShirt_type(delta.CopyAndAppendIndex(b.location, int64(i)))
}
func (b Shirt_list) Insert(index int, value *Shirt) *delta.Op {
	return delta.Insert(delta.CopyAndAppendIndex(b.location, int64(index)), value)
}
func (b Shirt_list) Move(from, to int) *delta.Op {
	return delta.Move(delta.CopyAndAppendIndex(b.location, int64(from)), int64(to))
}
func (b Shirt_list) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_list) Set(value []*Shirt) *delta.Op {
	return delta.Set(b.location, value)
}

type Shirt_bool_map struct {
	location []*delta.Locator
}

func (b Shirt_bool_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_bool_map(l []*delta.Locator) Shirt_bool_map {
	return Shirt_bool_map{location: l}
}
func (b Shirt_bool_map) Key(key bool) Shirt_type {
	return NewShirt_type(delta.CopyAndAppendKeyBool(b.location, key))
}
func (b Shirt_bool_map) Rename(from, to bool) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyBool(b.location, from), to)
}
func (b Shirt_bool_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_bool_map) Set(value map[bool]*Shirt) *delta.Op {
	return delta.Set(b.location, value)
}

type Shirt_int32_map struct {
	location []*delta.Locator
}

func (b Shirt_int32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_int32_map(l []*delta.Locator) Shirt_int32_map {
	return Shirt_int32_map{location: l}
}
func (b Shirt_int32_map) Key(key int) Shirt_type {
	return NewShirt_type(delta.CopyAndAppendKeyInt32(b.location, int32(key)))
}
func (b Shirt_int32_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyInt32(b.location, int32(from)), int32(to))
}
func (b Shirt_int32_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_int32_map) Set(value map[int32]*Shirt) *delta.Op {
	return delta.Set(b.location, value)
}

type Shirt_int64_map struct {
	location []*delta.Locator
}

func (b Shirt_int64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_int64_map(l []*delta.Locator) Shirt_int64_map {
	return Shirt_int64_map{location: l}
}
func (b Shirt_int64_map) Key(key int) Shirt_type {
	return NewShirt_type(delta.CopyAndAppendKeyInt64(b.location, int64(key)))
}
func (b Shirt_int64_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyInt64(b.location, int64(from)), int64(to))
}
func (b Shirt_int64_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_int64_map) Set(value map[int64]*Shirt) *delta.Op {
	return delta.Set(b.location, value)
}

type Shirt_uint32_map struct {
	location []*delta.Locator
}

func (b Shirt_uint32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_uint32_map(l []*delta.Locator) Shirt_uint32_map {
	return Shirt_uint32_map{location: l}
}
func (b Shirt_uint32_map) Key(key int) Shirt_type {
	return NewShirt_type(delta.CopyAndAppendKeyUint32(b.location, uint32(key)))
}
func (b Shirt_uint32_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyUint32(b.location, uint32(from)), uint32(to))
}
func (b Shirt_uint32_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_uint32_map) Set(value map[uint32]*Shirt) *delta.Op {
	return delta.Set(b.location, value)
}

type Shirt_uint64_map struct {
	location []*delta.Locator
}

func (b Shirt_uint64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_uint64_map(l []*delta.Locator) Shirt_uint64_map {
	return Shirt_uint64_map{location: l}
}
func (b Shirt_uint64_map) Key(key int) Shirt_type {
	return NewShirt_type(delta.CopyAndAppendKeyUint64(b.location, uint64(key)))
}
func (b Shirt_uint64_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyUint64(b.location, uint64(from)), uint64(to))
}
func (b Shirt_uint64_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_uint64_map) Set(value map[uint64]*Shirt) *delta.Op {
	return delta.Set(b.location, value)
}

type Shirt_string_map struct {
	location []*delta.Locator
}

func (b Shirt_string_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShirt_string_map(l []*delta.Locator) Shirt_string_map {
	return Shirt_string_map{location: l}
}
func (b Shirt_string_map) Key(key string) Shirt_type {
	return NewShirt_type(delta.CopyAndAppendKeyString(b.location, key))
}
func (b Shirt_string_map) Rename(from, to string) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyString(b.location, from), to)
}
func (b Shirt_string_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Shirt_string_map) Set(value map[string]*Shirt) *delta.Op {
	return delta.Set(b.location, value)
}

File: test/pb/tests2/tests2.op.dart
import 'package:protod/delta/delta.dart' as delta;
import 'package:protod/delta/delta.pb.dart' as delta;
import 'package:fixnum/fixnum.dart' as fixnum;
import 'package:protod/delta/delta.op.dart' as delta;
import 'tests2.pb.dart' as pb;

Op_root_type get op {
  return Op_root_type();
}

class Op_root_type {
  Op_root_type();
  Shirt_type get shirt {
    return Shirt_type([]);
  }
}

class Shirt_type extends delta.Location {
  Shirt_type(List<delta.Locator> location) : super(location);
  delta.String_scalar get designer {
    return delta.String_scalar(delta.copyAndAppendField(location, "designer", 1));
  }
  delta.Uint32_scalar get size {
    return delta.Uint32_scalar(delta.copyAndAppendField(location, "size", 2));
  }
  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set( value) {
    return delta.set(location, value);
  }

}

class Shirt_list extends delta.Location {
  Shirt_list(List<delta.Locator> location) : super(location);
  Shirt_type index(int i) {
    return Shirt_type(delta.copyAndAppendIndex(location, fixnum.Int64(i)));
  }
  delta.Op insert(int index, pb.Shirt value) {
    return delta.insert(delta.copyAndAppendIndex(location, fixnum.Int64(index)), value);
  }

  delta.Op move(int from, int to) {
    return delta.move(delta.copyAndAppendIndex(location, fixnum.Int64(from)), fixnum.Int64(to));
  }

  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set(List<pb.Shirt> value) {
    return delta.set(location, value);
  }

}

class Shirt_bool_map extends delta.Location {
  Shirt_bool_map(List<delta.Locator> location) : super(location);
  Shirt_type key(bool key) {
    return Shirt_type(delta.copyAndAppendKeyBool(location, key));
  }
  delta.Op rename(bool from, bool to) {
    return delta.rename(delta.copyAndAppendKeyBool(location, from), delta.keyBool(to));
  }

  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set(Map<bool, pb.Shirt> value) {
    return delta.set(location, value);
  }

}

class Shirt_int32_map extends delta.Location {
  Shirt_int32_map(List<delta.Locator> location) : super(location);
  Shirt_type key(int key) {
    return Shirt_type(delta.copyAndAppendKeyInt32(location, key));
  }
  delta.Op rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyInt32(location, from), delta.keyInt32(to));
  }

  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set(Map<int, pb.Shirt> value) {
    return delta.set(location, value);
  }

}

class Shirt_int64_map extends delta.Location {
  Shirt_int64_map(List<delta.Locator> location) : super(location);
  Shirt_type key(int key) {
    return Shirt_type(delta.copyAndAppendKeyInt64(location, fixnum.Int64(key)));
  }
  delta.Op rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyInt64(location, fixnum.Int64(from)), delta.keyInt64(fixnum.Int64(to)));
  }

  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set(Map<fixnum.Int64, pb.Shirt> value) {
    return delta.set(location, value);
  }

}

class Shirt_uint32_map extends delta.Location {
  Shirt_uint32_map(List<delta.Locator> location) : super(location);
  Shirt_type key(int key) {
    return Shirt_type(delta.copyAndAppendKeyUint32(location, key));
  }
  delta.Op rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyUint32(location, from), delta.keyUint32(to));
  }

  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set(Map<int, pb.Shirt> value) {
    return delta.set(location, value);
  }

}

class Shirt_uint64_map extends delta.Location {
  Shirt_uint64_map(List<delta.Locator> location) : super(location);
  Shirt_type key(int key) {
    return Shirt_type(delta.copyAndAppendKeyUint64(location, fixnum.Int64(key)));
  }
  delta.Op rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyUint64(location, fixnum.Int64(from)), delta.keyUint64(fixnum.Int64(to)));
  }

  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set(Map<fixnum.Int64, pb.Shirt> value) {
    return delta.set(location, value);
  }

}

class Shirt_string_map extends delta.Location {
  Shirt_string_map(List<delta.Locator> location) : super(location);
  Shirt_type key(String key) {
    return Shirt_type(delta.copyAndAppendKeyString(location, key));
  }
  delta.Op rename(String from, String to) {
    return delta.rename(delta.copyAndAppendKeyString(location, from), delta.keyString(to));
  }

  delta.Op delete() {
    return delta.delete(location);
  }

  delta.Op set(Map<String, pb.Shirt> value) {
    return delta.set(location, value);
  }

}

